.travis.yml

```diff
@@ -5,38 +5,159 @@ env:
     - secure: "JV6xOo1YVqUUlsa/za7BGuYZj//y5vg7Dgcn6v0MJzF+ZxIM5maGh6iA3mhiCIUetHVwwlqjJg8ZaZ8vFcWRcx27+591xgTSJWrPhgnbqcNupDCGoexiqUUzfgOtkuDowmRp4wz2e5uUpox4Gm6gmJkEt1nUs+/gsk7fjDoChSTGgJ2eASiLgzY2Hg38iT5IiZMx//FWw0uQnYCtPR0vtGYdgT8aSMXnVSsFqvpUJBUnQj4R93eJbQiXUN6SCRpjvd536JOgd+ud+zm7QTvzmRKSgag31XDKNjD/1SK3zzhkfNL/1T7Kr3rCdluxa0RO2c3eLiN6rkjelVtFcRvj7zwJvZ2BJ5WLWILtvbvvCmNzI43v+8RandYiWm8CWxVwUt3Cw+xzG0DblO+TCT8P9DipljI5zOeEBh6Tc920zb0P45E5jnxu6oxS2sOV/K2oyFu5152JXI4u24xVxMeVd9cxOzd4Jpxn5pe/m7RPeEDTpHA28RBTjA98kysR3aUBsfF5I/IrxkT8e9XnMuAOGaVgaAzGRAjL481H8CMIkQT14hO3+jqPKcn1scgDOGxw+Ijujdix9BtVt5/pEokfXYd2KXBmgDAPTBqfgkabPDXoNv0iXDdkQVl0924s0Jq/Io1DbjIIS+zezrPOwryVBsPjRYfWsjwl9Nv1KjK+yeQ="
 
+
+# enable c++11/14 builds
 addons:
   apt:
-    sources: [ 'ubuntu-toolchain-r-test', 'llvm-toolchain-trusty-4.0' ]
-    packages: [ 'libstdc++-4.9-dev', 'clang-4.0' ]
+    sources: [ 'ubuntu-toolchain-r-test' ]
+    packages: [ 'libstdc++-4.9-dev' ]
 
 install:
-  - npm install
+  - node -v
+  - which node
+  - clang++ -v
+  - which clang++
+  - make ${BUILDTYPE}
 
-script:
+# *Here we run tests*
+# We prefer running tests in the 'before_script' section rather than 'script' to ensure fast failure.
+# Be aware that if you use the 'script' section it will continue running all commands in the section even if one line fails.
+# This is documented at https://docs.travis-ci.com/user/customizing-the-build#Breaking-the-Build
+# We don't want this behavior because otherwise we might risk publishing builds when the tests did not pass.
+# For this reason, we disable the 'script' section below, since we prefer using 'before_script'.
+before_script:
   - npm test
-  - ./scripts/publish.sh
 
+script:
+  # after successful tests, publish binaries if specified in commit message
+  - ./scripts/publish.sh --TOOLSET=${TOOLSET:-} --debug=$([ "${BUILDTYPE}" == 'debug' ] && echo "true" || echo "false")
+
+# the matrix allows you to specify different operating systems and environments to 
+# run your tests and build binaries
 matrix:
   include:
+
+    ## ** Builds that are published **
+
+    # linux cfi build node v6/release
     - os: linux
-      node_js: 4
-      sudo: false
-      env: CC='clang-4.0' CXX='clang++-4.0'
+      env: BUILDTYPE=release TOOLSET=cfi CXXFLAGS="-fsanitize=cfi -fvisibility=hidden" LDFLAGS="-fsanitize=cfi"
+      node_js: 6
+    # linux publishable node v6/release
     - os: linux
+      env: BUILDTYPE=release
       node_js: 6
-      sudo: false
-      env: CC='clang-4.0' CXX='clang++-4.0'
+    # linux publishable node v6/debug
     - os: linux
-      node_js: 8
-      sudo: false
-      env: CC='clang-4.0' CXX='clang++-4.0'
+      env: BUILDTYPE=debug
```

Comment:
Debug builds need debug libortools otherwise they won't help at all

Reply:
In my experience there is still value to `Debug` builds (even if an external library is not also built with symbols) since `Debug` builds might trigger more warnings or uncover bugs in the node specific code that is compiled.

But yes, if we want a debug version of `libortools`, we could package that in mason (or this debug job could compile it from source on the fly?).